File: app/src/main/java/br/com/popularmovies/data/local/typeconverters/LocalDateConverter.java
package br.com.popularmovies.data.local.typeconverters;

import androidx.room.TypeConverter;

import org.joda.time.LocalDate;

public class LocalDateConverter {
    @TypeConverter
    public static LocalDate toLocalDate(String date) {
        return date == null ? null : LocalDate.parse(date);
    }

    @TypeConverter
    public static String toString(LocalDate date) {
        return date == null ? null : date.toString();
    }
}

File: app/src/main/java/br/com/popularmovies/core/network/local/typeconverters/LocalDateConverter.kt
package br.com.popularmovies.core.network.local.typeconverters

import androidx.room.TypeConverter

import org.joda.time.LocalDate

class LocalDateConverter {
    @TypeConverter
    fun toLocalDate(date: String?): LocalDate? {
        return date.let {
            LocalDate.parse(date)
        } ?: run { null }
    }

    @TypeConverter
    fun toString(date: LocalDate?): String? {
        return date?.toString()
    }
}
